Client/Frontend/Login Management/BreachAlertsManager.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  BreachAlertsManager.swift
+//  Client
+//
+//  Created by Vanna Phong on 5/21/20.
+//  Copyright © 2020 Mozilla. All rights reserved.
+//
+
+import Foundation
+import Storage // or whichever module has the LoginsRecord class
+
+/// Breach structure decoded from JSON
+struct BreachRecord: Codable {
+    var name: String
+    var title: String
+    var domain: String
+    var breachDate: String
+    var addedDate: String
+    var modifiedDate: String
+    var pwnCount: Int
+    var description: String
+    var logoPath: String
+    var isVerified: Bool
+    var isFabricated: Bool
+    var isSensitive: Bool
+    var isRetired: Bool
+    var isSpamList: Bool
+    var logoUrl: String?
+}
+
+/// A manager for the user's breached login information, if any.
+public class BreachAlertsManager {
+
+    //
+    // MARK: - Variables and Properties
+    //
+    var dataTask: URLSessionDataTask?
+    var endpointURL = "https://monitor.firefox.com/hibp/breaches"
+    var breaches: [BreachRecord] = []
+
+    //
+    // MARK: - Internal Methods
+    //
+    /// Loads breaches from Monitor endpoint.
+    public func loadBreaches() {
+        guard let url = URL(string: endpointURL) else {
+            return
+        }
+
+        dataTask?.cancel()
+
+        dataTask = URLSession.shared.dataTask(with: url) { data, response, error in
+            guard self.validatedHTTPResponse(response) != nil else {
+                print("loadBreaches(): invalid HTTP response")
+                return
+            }
+
+            if let error = error {
+                print("loadBreaches(): error: \(error)")
+                return
+            }
+
+            guard let data = data, !data.isEmpty else {
+                print("loadBreaches(): invalid data")
+                return
+            }
+
+            let decoder = JSONDecoder()
+
+            // .convertFromPascalCase
+            decoder.keyDecodingStrategy = .custom { keys in
+                let key = keys.last! // make sure array not empty
+
+                // string manipulation
+                let keyStr = key.stringValue
+                let pascalToCamel = keyStr.prefix(1).lowercased() + keyStr.dropFirst()
+
+                // CodingKey conformity
+                let codingKeyType = type(of: key)
+                return codingKeyType.init(stringValue: pascalToCamel)!
+
```

Comment:
You can set custom keys for the codable variables if their names don't match to the keys in the data you get back. I think that would be ideal here.

Reply:
All of the endpoint keys are in Pascal case, so I was thinking along the lines of using the Swift-given `decoder.keyDecodingStrategy = .convertFromSnakeCase` except there is no .convertFromPascalCase so I made a custom one. Maybe this block of code is better suited to an extension though